File: engine/src/main/java/com/bartlomiejpluta/base/engine/database/service/DatabaseService.java
package com.bartlomiejpluta.base.engine.database.service;

import com.bartlomiejpluta.base.engine.common.init.Initializable;
import com.bartlomiejpluta.base.internal.gc.Cleanable;

import java.sql.Connection;

public interface DatabaseService extends Initializable, Cleanable {
   Connection getConnection();
}

File: editor/src/main/kotlin/com/bartlomiejpluta/base/editor/database/service/DatabaseService.kt
package com.bartlomiejpluta.base.editor.database.service

import com.bartlomiejpluta.base.editor.database.model.data.Query
import com.bartlomiejpluta.base.editor.database.model.schema.SchemaDatabase
import com.bartlomiejpluta.base.editor.database.model.schema.SchemaTable
import java.sql.Connection

interface DatabaseService {
   val database: SchemaDatabase

   fun <T> run(op: Connection.() -> T): T?

   fun execute(statement: String, name: String, schema: SchemaTable? = null): Query?

   fun dump()
}